build.gradle

```diff
@@ -1,3 +1,320 @@
-plugins {
-    id 'java-base'
+buildscript {
+    repositories {
+        jcenter()
+    }
+    dependencies {
+        classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.5'
+        classpath 'com.google.gradle:osdetector-gradle-plugin:1.6.0'
+        classpath 'com.github.jengelman.gradle.plugins:shadow:2.0.4'
+        classpath files('gradle/witness/gradle-witness.jar')
+        classpath 'org.springframework.boot:spring-boot-gradle-plugin:1.5.10.RELEASE'
+    }
+}
+
+
+configure(subprojects) {
+    apply plugin: 'java'
+    apply plugin: 'com.google.osdetector'
+
+    sourceCompatibility = 1.10
+
+    ext {
+        bcVersion = '1.56'
+        codecVersion = '1.9'
+        easyVersion = '4.0.1'
+        jmockitVersion = '1.42'
+        joptVersion = '5.0.3'
+        junitVersion = '4.12'
+        langVersion = '3.4'
+        libdohjVersion = 'd4ace7bc'
+        lombokVersion = '1.18.2'
+        mockitoVersion = '2.21.0'
+        powermockVersion = '2.0.0-beta.5'
+        protobufVersion = '3.5.1'
+        sparkVersion = '2.5.2'
+        springVersion = '4.3.6.RELEASE'
+
+        os = osdetector.os == 'osx' ? 'mac' : osdetector.os == 'windows' ? 'win' : osdetector.os
+    }
+
+    repositories {
+        jcenter()
+        maven { url 'https://jitpack.io' }
+        maven { url 'https://raw.githubusercontent.com/JesusMcCloud/tor-binary/master/release/' }
+    }
+
+    dependencies {
+        testCompile 'junit:junit:4.12'
+    }
+
+    tasks.withType(JavaCompile) {
+        options.encoding = 'UTF-8'
+    }
+}
+
+
+configure([project(':desktop'),
+           project(':monitor'),
+           project(':relay'),
+           project(':seednode'),
+           project(':statsnode')]) {
+    apply plugin: 'application'
+
+    build.dependsOn installDist
+    installDist.destinationDir = file('build/app')
+    distZip.enabled = false
+}
+
+
+configure(project(':assets')) {
+    dependencies {
+        compile("network.bisq.libdohj:libdohj-core:$libdohjVersion") {
+            exclude(module: 'protobuf-java')
+        }
+        compile "commons-codec:commons-codec:$codecVersion"
+        compile "org.apache.commons:commons-lang3:$langVersion"
+        compile "org.bouncycastle:bcpg-jdk15on:$bcVersion"
+    }
+}
+
+
+configure(project(':common')) {
+    apply plugin: 'com.google.protobuf'
+
+    sourceSets.main.java.srcDir "$buildDir/generated/source/proto/main/java"
+
+    protobuf {
+        protoc {
+            artifact = "com.google.protobuf:protoc:$protobufVersion"
+        }
+    }
+
+    dependencies {
+        compile "org.openjfx:javafx-base:11:$os"
+        compile "org.openjfx:javafx-graphics:11:$os"
+        compile "com.google.protobuf:protobuf-java:$protobufVersion"
+        compile 'com.google.code.gson:gson:2.7'
+        compile('com.googlecode.json-simple:json-simple:1.1.1') {
+            exclude(module: 'junit')
+        }
+        compile 'org.springframework:spring-core:4.3.6.RELEASE'
```

Comment:
Since you have the spring version defined above, I assume this can be changed to the following?
`compile "org.springframework:spring-core:$springVersion"`

Reply:
Yes, an oversight on my part, thanks. Fixed in 978fd2d4a.